common/v2/utils/useTxMulti/reducer.ts

```diff
@@ -0,0 +1,134 @@
+  status: ITxStatus.PREPARING
+});
+
+export function TxMultiReducer(state: TxMultiState, action: TxMultiAction): TxMultiState {
+  const { type, payload, error } = action;
+  switch (type) {
+    case ActionTypes.INIT_REQUEST: {
+      return {
+        ...state,
+        isSubmitting: true
+      };
+    }
+    case ActionTypes.INIT_SUCCESS: {
+      const { txs, account, network } = payload;
+      return {
+        ...state,
+        _isInitialized: true,
+        canYield: true,
+        transactions: R.map(formatTx, txs),
+        isSubmitting: false,
+        account,
+        network
+      };
+    }
+    case ActionTypes.INIT_FAILURE: {
+      return {
+        ...state,
+        isSubmitting: false,
+        ...(error && { error: payload })
+      };
+    }
+    case ActionTypes.PREPARE_TX_REQUEST: {
+      const transactions = R.adjust(
+        state._currentTxIdx,
+        R.mergeLeft({ status: ITxStatus.PREPARING }),
+        state.transactions
+      );
+      return {
+        ...state,
+        isSubmitting: true,
+        transactions
+      };
+    }
+    case ActionTypes.PREPARE_TX_SUCCESS: {
+      const { txRaw } = payload;
+      const transactions = R.adjust(
+        state._currentTxIdx,
+        R.mergeLeft({ txRaw, status: ITxStatus.READY }),
+        state.transactions
+      );
+      return {
+        ...state,
+        _isInitialized: true,
+        canYield: true,
+        isSubmitting: false,
+        transactions
+      };
+    }
+    case ActionTypes.PREPARE_TX_FAILURE: {
+      return {
+        ...state,
+        ...(error && { error: payload })
+      };
+    }
+    case ActionTypes.SEND_TX_SUCCESS: {
+      const { txHash } = payload;
+      const transactions = R.adjust(
+        state._currentTxIdx,
+        R.mergeLeft({ txHash, status: ITxStatus.BROADCASTED }),
+        state.transactions
+      );
+
+      return {
+        ...state,
+        transactions,
+        isSubmitting: false,
+        canYield: true
+      };
+    }
+    case ActionTypes.CONFIRM_TX_REQUEST: {
+      return {
+        ...state,
+        isSubmitting: true
+      };
+    }
+    case ActionTypes.CONFIRM_TX_SUCCESS: {
+      const { txReceipt, minedAt } = payload;
+      const next = (curr: number) => Math.min(curr + 1, state.transactions.length - 1);
+      const transactions = R.adjust(
+        state._currentTxIdx,
+        R.mergeLeft({ txReceipt, minedAt, status: ITxStatus.CONFIRMED }),
+        state.transactions
+      );
+
+      return {
+        ...state,
+        _currentTxIdx: next(state._currentTxIdx),
+        transactions
+      };
+    }
+    case ActionTypes.CONFIRM_TX_FAILURE: {
+      return {
+        ...state,
+        isSubmitting: false,
+        ...(error && { error: payload })
+      };
+    }
+    case ActionTypes.HALT_FLOW: {
+      return { ...state, canYield: false };
+    }
+    case ActionTypes.RESET: {
+      return initialState;
+    }
+    default:
+      return state;
+  }
```

Comment:
Can you add the logic to add these TXs to LS `account.transactions` array on `PENDING | SUCCESS | FAILED` ?

Reply:
Hmm, after looking into it do you mind if we tackle this later?
Currently we're storing way too much information within the account.transactions which has a `senderAccount` that seems to be full blow `StoreAccount`, and it seems good to tackle it in one blow.

Replace all our send transactions with this hook, have a single place to `addNewTranscation`, and handle the polling when the transactions are updated.

Reply:
> Currently we're storing way too much information within the account.transactions which has a `senderAccount`

I believe this `senderAccount` only exists while the tx is pending, then it's overwritten with the tx receipt when it's done.

> Currently we're storing way too much information within the account.transactions

But agree. 

> after looking into it do you mind if we tackle this later?

So no, i don't mind.